package/MDAnalysis/analysis/rms.py

```diff
@@ -310,6 +309,20 @@ def process_selection(select):
     select['reference'] = asiterable(select['reference'])
     return select
 
+def weight_type_check(weights,atoms,selection):
+
+    if (np.array(weights).ndim == 1) & (np.array(weights).dtype 
+                            in (np.dtype('float64'),np.dtype('int64'))):
+        if len(weights) != atoms.n_atoms:
+            raise ValueError("Length of provided weights {} do not match the number of atoms "
+                "in the selection {}: {}".format(weights, selection['mobile'], atoms.n_atoms))
+        else:
+            tmp_weights = get_weights(atoms, weights) 
```

Comment:
Remove the `tmp_weights` and just call the get_weights(atoms, weights) to check wether your weights are okay. 

Maybe there is also a way to combine your method with `get_weights` from utils.